### DISCUSSION TITLE
How to keep the resolvers DRY?

### DISCUSSION THREAD

**Discussion by alucardu** [prisma/prisma#9125] [Q&A] [answered]
I have 1 query and 2 mutations that all return an array of records:

```ts
  moviesFromUser: async (root, args, {res, req}) => {
    if (req.userId) {
      const user = await prisma.user.findMany({
        where: {id: args.userId || req.userId},
        select: {
          movies: true,
        },
      });
      return user[0].movies;
    }
  },
  ```
  
  ```
    addUserToMovie: async (_, args, {req, res}) => {
    const moviesAlreadyExists = await prisma.movie.findUnique({
      where: {tmdb_id: args.tmdb_id}});

    if (!moviesAlreadyExists) {
      ...
    }

    const user = await prisma.user.findMany({
      where: {id: req.userId},
      select: {
        movies: true,
      },
    });

    return user[0].movies;
  },
  ```
  
  ```ts
    removeMovie: async (parent, args, {req, res}) => {
    await prisma.user.update({
      where: {id: req.userId},
      data: {
        movies: {
          disconnect: [{id: args.id}],
        },
      },
    });

    const user = await prisma.user.findMany({
      where: {id: req.userId},
      select: {
        movies: true,
      },
    });

    return user[0].movies;
  },
  ```

as you might have noticed I repeat this code:

```ts
const user = await prisma.user.findMany({
      where: {id: req.userId},
      select: {
        movies: true,
      },
    });
```

3 times. It would be much easier and cleaner (I think) if I could reference the `moviesFromUser` query from the mutations to return the records.

**Comment by ryands17** [comment, marked as answer]
@alucardu 👋 

Extracting it to a function and referencing that is one way you could do it.

**Reply by alucardu** [reply]
Oh I can't reference the query directly but i can wrap it in a function that returns the requested data. Great.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
